### PR TITLE
Don't show own user in search results

### PR DESCRIPTION
**Pull request by csmith**
Anywhere we do a global search (starting a DM, creating a room, inviting a user to a room) should filter out the local user.

### CODE REVIEW

File: libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt [hunk 1]
```diff
@@ -31,24 +32,31 @@ import javax.inject.Inject
...
 ) : UserRepository {
 
+    private val sessionId = client.sessionId
+
     override suspend fun search(query: String): Flow<List<UserSearchResult>> = flow {
-        // Manually add a fake result with the matrixId, if any
-        val isUserId = MatrixPatterns.isUserId(query)
-        if (isUserId) {
+        // Manually add a fake result with the matrixId, if provided and not the local user
+        val isAnotherUsersId =  query != sessionId.value && MatrixPatterns.isUserId(query)
```
Comment (ganfra) [inline libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt:43]: Maybe create a MatrixClient.isMe(userId: UserId): Boolean extension method and check for this already existent usage?

**Review by ganfra — APPROVED** [review]
Just one remark otherwise LGTM

File: libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt (hunk 1 above)
Comment (bmarty) [inline libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt:43]: `isAnotherUsersId` -> `isAnotherUserId`, and please format the file :)

Comment (csmith) [reply libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt:43]: It was meant to read like "is another user's id", but I can see the ambiguity without punctuation :laughing:. I don't think `isAnotherUserId` quite conveys what it means, so I've changed it entirely to `shouldQueryProfile`.

**Pull request merged**